api/v1/csi_node_driver.go

```diff
@@ -0,0 +1,148 @@
+	// Affinity is a group of affinity scheduling rules for the csi-node-driver pods.
+	// If specified, this overrides any affinity that may be set on the csi-node-driver DaemonSet.
+	// If omitted, the csi-node-driver DaemonSet will use its default value for affinity.
+	// WARNING: Please note that this field will override the default csi-node-driver DaemonSet affinity.
+	// +optional
+	Affinity *v1.Affinity `json:"affinity"`
+
+	// NodeSelector is the csi-node-driver pod's scheduling constraints.
+	// If specified, each of the key/value pairs are added to the csi-node-driver DaemonSet nodeSelector provided
+	// the key does not already exist in the object's nodeSelector.
+	// If omitted, the csi-node-driver DaemonSet will use its default value for nodeSelector.
+	// WARNING: Please note that this field will modify the default csi-node-driver DaemonSet nodeSelector.
+	// +optional
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+
+	// Tolerations is the csi-node-driver pod's tolerations.
+	// If specified, this overrides any tolerations that may be set on the csi-node-driver DaemonSet.
+	// If omitted, the csi-node-driver DaemonSet will use its default value for tolerations.
+	// WARNING: Please note that this field will override the default csi-node-driver DaemonSet tolerations.
+	// +optional
+	Tolerations []v1.Toleration `json:"tolerations"`
+}
+
+// CSINodeDriverDaemonSetPodTemplateSpec is the csi-node-driver DaemonSet's PodTemplateSpec
+type CSINodeDriverDaemonSetPodTemplateSpec struct {
+	// Metadata is a subset of a Kubernetes object's metadata that is added to
+	// the pod's metadata.
+	// +optional
+	Metadata *Metadata `json:"metadata,omitempty"`
+
+	// Spec is the csi-node-driver DaemonSet's PodSpec.
+	// +optional
+	Spec *CSINodeDriverDaemonSetPodSpec `json:"spec,omitempty"`
+}
+
+// CSINodeDriverDaemonSet is the configuration for the csi-node-driver DaemonSet.
+type CSINodeDriverDaemonSet struct {
+	// Metadata is a subset of a Kubernetes object's metadata that is added to the DaemonSet.
+	// +optional
+	Metadata *Metadata `json:"metadata,omitempty"`
+
+	// Spec is the specification of the csi-node-driver DaemonSet.
+	// +optional
+	Spec *CSINodeDriverDaemonSetSpec `json:"spec,omitempty"`
+}
+
+// CSINodeDriverDaemonSetSpec defines configuration for the csi-node-driver DaemonSet.
+type CSINodeDriverDaemonSetSpec struct {
+	// Template describes the csi-node-driver DaemonSet pod that will be created.
+	// +optional
+	Template *CSINodeDriverDaemonSetPodTemplateSpec `json:"template,omitempty"`
+}
+
+func (c *CSINodeDriverDaemonSet) GetMetadata() *Metadata {
+	return c.Metadata
+}
+
+func (c *CSINodeDriverDaemonSet) GetMinReadySeconds() *int32 {
+	return nil
+}
+
+func (c *CSINodeDriverDaemonSet) GetPodTemplateMetadata() *Metadata {
+	if c.Spec != nil {
+		if c.Spec.Template != nil {
+			return c.Spec.Template.Metadata
+		}
+	}
+	return nil
+}
+
+func (c *CSINodeDriverDaemonSet) GetInitContainers() []v1.Container {
```

Comment:
Please add a comment here indicating that the CSI DaemonSet does not contain any init containers so that we know the lack of implementation was deliberate.

Reply:
thank you for review, applied

Reply:
> Please add a comment here indicating that the CSI DaemonSet does not contain any init containers so that we know the lack of implementation was deliberate.

thank you for review, applied